sdk/storage/azure-storage-blob/azure/storage/blob/_quick_query_helper.py

```diff
@@ -27,25 +31,57 @@ class QuickQueryReader(object):  # pylint: disable=too-many-instance-attributes
 
     def __init__(
         self,
-        client=None,
         name=None,
         container=None,
-        progress_callback=None,
+        errors='strict',
         encoding=None,
-        **kwargs
+        headers=None,
+        response=None
     ):
         self.name = name
         self.container = container
-        self.response_headers = None
-        self.total_bytes = None
+        self.response_headers = headers
+        self.size = None
         self.bytes_processed = 0
```

Comment:
now self.bytes_processed is the only place user can get the query progress.
So user need to do know the download progress by polling the attribute on blob_query_reader like this, and it's possible the result is like 5, 10, 10, 10, 15, 15, 15
```
while True:  
    print(blob_query_reader.bytes_processed)
```
So I still feel we should provide a callback function to let the user print the progress, do you have any thought? @kasobol-msft

Reply:
I would have a sample snippet as something like this:
```
for record in query_reader.records():
    print("Processed {} bytes of total {}".format(query_reader.bytes_processed, query_reader.size))
    do_someting(record)
```
This is a more Pythonic approach to handling progress in Python.
However if we get feedback that there's a strong need to support a callback - this can be added as a non-breaking update at any time.

Reply:
the snippet will print the results like 1000, 1000, 1000, 1000, 1000 if there are 5 records in the arvo data.
bytes_processed is updated (`self.bytes_processed = result.get('bytesScanned')` ) when the bunch of records are returned, not per record.

Reply:
I agree with @xiafu-msft - it's going to look odd. 
As far as I understand in case of `download` we can pass a callback that listens to each chunked response (afaik through `raw_response_hook`).

On other platforms (c#,java) we solve this problem for both query and downloads by letting customer pass a callback. These callbacks are explicit parameters for query and donwload methods. I'm surprised to see similar solution here (and for download).

Reply:
My resistance here is based on callbacks are not a common convention in Python. Most standard Python HTTP libraries don't provide any kind of progress callback.
I agree that if we're going to support a callback-pattern - it should be consistent with that used by download - and I discussed this earlier with @xiafu-msft.
There are some restrictions here regarding the generated code - however I will revisit and see if I can get that working.

Reply:
I think the the reason was there's only one response for quick query, so we cannot use the raw_response_hook in the pipeline. For download progress, the callback is executed every time there's a response sent back

Reply:
If we're unsure how to expose progress reporting now then I'd recommend to not implement it for now and add it later once we have better story for that. Maybe nobody is going to ask for it.

Reply:
That sounds good! @kasobol-msft

Reply:
shall we hide self.bytes_processed and self.size, since they are not indicating the processed_bytes/size of selected records.

Reply:
Thanks @kasobol-msft, @xiafu-msft -I think that's a great plan :)
I will make the updates - and remove those two properties.